### ISSUE TITLE
Project works fine on Windows computer, but gets stuck without output when moved to cluster

### ISSUE THREAD

**Issue by FAOlivieri** [DeepLabCut/DeepLabCut#3267]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating System

NAME="Rocky Linux"
VERSION="8.10 (Green Obsidian)"

### DeepLabCut version

Tested on 3.0.0rc13 (installed in a conda environment) and 3.0.0rc14 (As a Docker file)

### What engine are you using?

pytorch

### DeepLabCut mode

single animal

### Device type

Tested on Nvidia L40S and Nvidia A40

### Bug description 🐛

I am using deeplabcut to analyze some videos of mice in a cage. I can make the whole project run correctly on windows. However, when I move my project to my linux cluster, I can't make it work.

After manually labeling the images I moved the entire folder to the cluster and changed the project_path in the config.yaml but whatever else I do, deeplabcut gets stuck without giving any input at all but also without erroring.

If I make a script starting with check_labels, my output only contains "Creating images with labels by [user]" and gets stuck there forever. I let it run for days without it generating the images.

Since checking labels is optional (and it worked on windows so I know the labels are fine) I tried running a script beginning with "create_training_dataset" but I get the output up until "Create training dataset" and the nothing else happens. The only output file I get, within training-datasets/iteration-0/UnaugmentedDataSet_TestOpenFieldBL6Mar30  is the file "metadata.yaml" but it only contains "shuffle: {}"   (while in the windows output I can see the correct content on this file). Again, no errors and the script continues for days without doing anything.

I made my python script print torch.cuda.is_available() and torch.cuda.device_count() and I get the correct output, so the script is able to detect my GPUs. If I run on a CPU node I get the same behaviour.

Like I mentioned, I've tried using both the docker version and the conda-installed version and they both behave the same way. 

### Steps To Reproduce

I attach here my config file and (just in case it's useful) my csv generated by data labeling. If needed I can provide the entire dataset

[config.yaml](https://github.com/user-attachments/files/26381390/config.yaml)

[CollectedData_Tjasa.csv](https://github.com/user-attachments/files/26381395/CollectedData_Tjasa.csv)

### Relevant log output

```shell
Importing
Loading DLC 3.0.0rc13...
DLC loaded in light mode; you cannot use any GUI (labeling, relabeling and standalone GUI)
3.0.0rc13
=== PyTorch / CUDA diagnostics ===
CUDA available: True
Device count: 1
CUDA_VISIBLE_DEVICES: 0
Torch version: 2.6.0+cu124
Torch CUDA version: 12.4
Current device: 0
Device name: NVIDIA L40S
GPU memory allocated: 0
GPU memory reserved: 0
==================================
Starting
Creating images with labels by Tjasa.
```

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by C-Achard** [comment]
Hello @FAOlivieri,

Thanks for opening this issue. From what you shared, this is likely due to the h5 containing outdated paths from Windows after moving.

If you could share:
- The first few entries of the index from the H5 (`pd.read_hdf(...).index[:10]`) on the cluster
- Or even better, the contents of `labeled-data/cutm3v1mp4/` (the one used on the cluster as well)

that would help confirm whether this is a path-mismatch issue or something else.

Thanks !

Best,
Cyril

**Comment by FAOlivieri** [comment]
Hi. Thank you for answering so quickly. This is the first 10 lines of the h5 file:

MultiIndex([('labeled-data', 'cutm3v1mp4', 'img0025.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0063.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0096.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0101.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0119.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0132.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0146.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0156.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0176.png'),
            ('labeled-data', 'cutm3v1mp4', 'img0185.png')],
           )

It matches perfectly the names of the directories and folders. The content of that folder is just the h5 file, the csv file I uploaded before and the images themselves

**Comment by C-Achard** [comment]
Thanks, that all looks fine, so it's most likely not an OS transferability issue.

One thing that could go wrong on a cluster is actually reading the files: are they stored on the compute node, or in a separate node that requires network for data read/write ? I'm not familiar with your current cluster architecture, but I can see that being an issue.

Copying the project to the node-local storage might help solve this, as it would eliminate any issues tied to network/permission/data IO limits across cluster filesystems. 

To check if the issue is from reading files, you could check if something basic such as :
```python
from pathlib import Path
from PIL import Image
import pandas as pd

project_path = Path("/home/tlapanja/Test_Fede/Data/testComp2/TestOpenFieldBL6-Tjasa-2026-03-30")
h5 = project_path / "labeled-data" / "cutm3v1mp4" / "CollectedData_Tjasa.h5"

df = pd.read_hdf(h5)
paths = [project_path.joinpath(*idx) for idx in df.index]

for p in paths:
    print("Checking:", p)
    with Image.open(p) as im:
        im.verify()

for p in paths[:10]:
    print(p, p.exists(), p.stat().st_size)

print("Done checking.")
```
also hangs or successfully runs.

If this runs fine, interrupting the process and checking where it is stuck (probably done already in cluster logs from the failed jobs?) would be very helpful. If not available, we can try setting up a faulthandler to get more specific info as to where it gets stuck exactly, let me know.

**Comment by FAOlivieri** [comment]
You seem to have hit the nail right on the head. I ran the script you gave me through SLURM  (adding print statements in each line) and it hangs at the step for reading the h5 file. Running the script manually on the headnode works fine.

Our cluster consists of a Headnode into which we login which then submits jobs to the compute nodes. This is the first time I encounter problems by just putting my files on the headnode. I am not sure why this is happening specifically with this h5 file (or the h5 reading method in pandas). Could you provide some guidance on how to fix this?

**Comment by C-Achard** [comment]
Thanks, that narrows it down a lot.
This looks much more like a cluster filesystem / HDF5 locking issue rather than a DLC issue. 
I'm not an expert, but it seems PyTables/HDF5 uses file locking, and this can hang on some shared/NFS-style HPC filesystems. 

Two things to try:
- Copy the H5 (or the whole project) to node-local file system and run from there,
- In the script, set export HDF5_USE_FILE_LOCKING=FALSE before starting Python (safe only if the file is not accessed concurrently, e..g by other jobs, make sure to back up the files)

At this point I’d also recommend checking with your cluster admins, simply mentioning that a basic `pd.read_hdf( )` call works on the head node but hangs on the compute node for the same file/path; since now it looks like a cluster filesystem issue rather than something specific to DLC. Your cluster might also have dedicated data nodes, perhaps worth checking.

Hope this helps!

Best,
Cyril

**Comment by C-Achard** [comment]
Hello @FAOlivieri,

I will close this for now since it is related to the cluster rather than DLC itself. 

Feel free to open a new issue if you face any other problem with DLC code! Thanks.
